crates/environ/src/component/vmcomponent_offsets.rs

```diff
@@ -114,9 +122,10 @@ impl<P: PtrSize> VMComponentOffsets<P> {
 
         fields! {
             size(magic) = 4u32,
-            size(flags) = 1u32,
             align(u32::from(ret.ptr.size())),
             size(store) = cmul(2, ret.ptr.size()),
+            size(flags) = ret.num_runtime_component_instances,
```

Comment:
better not to open code / bake in the assumption that the flags will always be one byte large, especially since we wrap them up in a `VMComponentFlags` data abstraction elsewhere:

```suggestion
            size(flags) = cmul(size_of_vm_component_flags(), ret.num_runtime_component_instances),
```